spec/facility_spec.rb

```diff
@@ -1,26 +1,156 @@
+         expect(@facility_1.collected_fees).to eq(200)
+     end
+  end
+
+  describe '#administer written test' do
+
+      before do
+          @facility_1.add_service('Written Test')
+      end
+
+      it 'can administer a written test' do
+          registrant_1 = Registrant.new('Bruce', 18, true )
+          expect(registrant_1.license_data[:written]).to eq(false)
+
+          @facility_1.administer_written_test(registrant_1)
+
+          expect(registrant_1.license_data[:written]).to eq(true)
+      end
+
+      it 'only administers written tests for registrants with a permit and are at least 16' do
+          registrant_3 = Registrant.new('Tucker', 15 )
+          expect(registrant_3.license_data[:written]).to eq(false)
+
+          @facility_1.administer_written_test(registrant_3)
+
+          expect(registrant_3.license_data[:written]).to eq(false)
+      end
+  end
+
+  describe '#administer a road test' do
+      it 'can administer a road test to those who have passed a written test' do
+          registrant_1 = Registrant.new('Bruce', 18, true )
+          @facility_1.add_service('Written Test')
+          @facility_1.add_service('Road Test')
+          expect(registrant_1.license_data[:written]).to eq(false)
+
+          @facility_1.administer_written_test(registrant_1)
+          expect(registrant_1.license_data[:written]).to eq(true)
+          @facility_1.administer_road_test(registrant_1)
+
+          expect(registrant_1.license_data[:license]).to eq(true)
+      end
+  end
+
+  describe '#permit necessary for written test' do
+
+      before do
+          @facility_1.add_service('Written Test')
+          @facility_1.add_service('Road Test')
+          @facility_1.add_service('Renew License')
+          @registrant_2 = Registrant.new('Penny', 16)
+      end
+
+     it 'can only give written tests if registrant has a permit' do
+         @facility_1.administer_written_test(@registrant_2)
+         expect(@registrant_2.permit?).to eq(false)
+         expect(@registrant_2.license_data[:written]).to eq(false)
+
+         @registrant_2.earn_permit
+         expect(@registrant_2.permit?).to eq(true)
+
+         @facility_1.administer_written_test(@registrant_2)
+         expect(@registrant_2.license_data[:written]).to eq(true)
+     end
+  end
+
+  describe '#license renewal' do
+      it 'can renew a license' do
+          registrant_1 = Registrant.new('Bruce', 18, true )
+          @facility_1.administer_written_test(registrant_1)
+          @facility_1.administer_road_test(registrant_1)
+
+          expect(registrant_1.license_data[:renewed]).to eq(false)
+
+          @facility_1.renew_license(registrant_1)
+
+          expect(registrant_1.license_data[:renewed]).to eq(true)
+      end
+
+      it 'will not renew if there is no license' do
+          registrant_2 = Registrant.new('Penny', 16)
+
+          @facility_1.renew_license(registrant_2)
+
+          expect(registrant_2.license_data[:renewed]).to eq(false)
+      end
```

Comment:
Excellent testing! I appreciate your use of describe blocks to categorize the tests.